compiler/noirc_frontend/src/parser/parser/expression.rs

```diff
@@ -109,15 +136,19 @@ impl Parser<'_> {
         }
     }
 
-    /// AtomOrUnaryRightExpression
-    ///     = Atom UnaryRightExpression*
-    fn parse_atom_or_unary_right(&mut self, allow_constructors: bool) -> Option<Expression> {
+    /// AtomOrMemberAccessOrCallOrMethodCallOrIndexExpression
+    ///     = Atom MemberAccessOrCallOrMethodCallOrIndexExpression*
+    fn parse_atom_or_member_access_or_call_or_method_call_or_index(
+        &mut self,
+        allow_constructors: bool,
+    ) -> Option<Expression> {
         let start_location = self.current_token_location;
         let mut atom = self.parse_atom(allow_constructors)?;
         let mut parsed;
 
         loop {
-            (atom, parsed) = self.parse_unary_right(atom, start_location);
+            (atom, parsed) =
+                self.parse_member_access_or_call_or_method_call_or_index(atom, start_location);
```

Comment:
Can't comment on it above since it wasn't changed but we can change `parse_atom` to `parse_quark` since I already used atom above.

Atom is a standard name, quark is not but it continues the theme and makes sense as an indivisible parselet

Reply:
In the end I went with `atom_rhs`